packages/kbn-alerts-ui-shared/src/common/hooks/use_load_connectors.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { useQuery } from '@tanstack/react-query';
+import type { HttpStart } from '@kbn/core-http-browser';
+import { fetchConnectors } from '../apis';
+
+export interface UseLoadConnectorsProps {
+  http: HttpStart;
+  includeSystemActions?: boolean;
+}
+
+export const useLoadConnectors = (props: UseLoadConnectorsProps) => {
+  const { http, includeSystemActions = false } = props;
+
+  const queryFn = () => {
+    return fetchConnectors({ http, includeSystemActions });
+  };
+
+  const { data, isLoading, isFetching, isInitialLoading } = useQuery({
```

Comment:
I might not have full idea here so might be stupid question: how do we handle error scenarios for all these hooks?

Reply:
Haha yea we don't really handle errors because if the endpoint fails then something quite wrong has happened